basics/holder.ts

```diff
@@ -7,8 +7,10 @@ import { Types } from '@kiltprotocol/credentials'
 export async function createCredentialPresentation(
   credential: Types.VerifiableCredential[],
   holderDid: DidDocument,
-  signers: SignerInterface[]
-): Promise<void> {
+  signers: SignerInterface[],
+  issuerDid: DidDocument
+): Promise<{ presentation: Types.VerifiablePresentation; challenge: string }> {
+  // Create a presentation
   const challenge = randomAsHex()
```

Comment:
FYI, using a challenge is optional, but you'd normally want to have either a challenge-response flow or a fairly short validUntil window to ensure that presentations are 'fresh' and not re-used, to avoid replays. Using domain- or verifier-bound presentations also helps to avoid re-use/replay of presentations in other contexts.

Example: I publish my credentials on my web-server, and do so by wrapping them in a presentation. This presentation has a longer time-to-live so I don't have to update it all the time, but is bound to the URL of my website using the `domain` field.